### PR TITLE
Incorporating generalized scripts into data-reconciliation-app

### LINKED ISSUE #76: Configure & test generalized governance scripts for data reconciliation app
Leverage the generic governance scripts for our skeleton app.
For the skeleton app, we can assume 1 member & 1 user for now...

1. Define config for skeleton app
2. Test adding a member, a user and proposing the skeleton app in enclave/virtual environment (not MCCF yet).

### PR DESCRIPTION
**Pull request by julioalex-rezende**
Fixes #76 

Including `make test`, `make test-docker-virtual` and `make test-docker-enclave` flows to our data-reconciliation-app, to reflect the most recent changes related to the generalization of scripts.

Now, `data-reconciliation-app` follows the same flow as `banking-app` (and future applications)

Given we still have some differences in folder structure for each app, I've overcame this by considering which location constitutions and certificate files are by requiring them as parameters when running locally (for sandbox script). This gives a higher flexibility to applications to organize as preferred in some cases.

### CODE REVIEW

File: data-reconciliation-app/Makefile [hunk 1]
```diff
@@ -10,25 +10,33 @@ build: ## 🔨 Build the Application
...
-	@./test/run_test.sh
+	@. ../scripts/test_sandbox.sh --nodeAddress 127.0.0.1:8000 --certificate_dir ./workspace/sandbox_common --constitution_dir ./governance/constitution
+
+test-docker-virtual: build-virtual ## 🧪 Test the Banking Application in a Docker sandbox
+	@echo -e "\e[34m$@\e[0m" || true
+	@. ../scripts/test_docker.sh --virtual --serverIP 172.17.0.3 --port 8080
+
+test-docker-enclave: build-enclave ## 🧪 Test the Banking Application in a Docker enclave
+	@echo -e "\e[34m$@\e[0m" || true
+	@. ./scripts/test_docker.sh --enclave --serverIP 172.17.0.4 --port 8080
```
Comment (ross-p-smith) [inline data-reconciliation-app/Makefile:23]:
```suggestion
	@. ../scripts/test_docker.sh --enclave --serverIP 172.17.0.4 --port 8080
```

File: data-reconciliation-app/Makefile [hunk 2]
```diff
@@ -10,25 +10,33 @@ build: ## 🔨 Build the Application
 	@echo -e "\e[34m$@\e[0m" || true
 	@npm run build
 
-test: ## 🧪 Run tests, used for local development
+test: build ## 🧪 Test the Banking Application in the sandbox
 	@echo -e "\e[34m$@\e[0m" || true
-	@./test/run_test.sh
+	@. ../scripts/test_sandbox.sh --nodeAddress 127.0.0.1:8000 --certificate_dir ./workspace/sandbox_common --constitution_dir ./governance/constitution
+
+test-docker-virtual: build-virtual ## 🧪 Test the Banking Application in a Docker sandbox
```
Comment (ross-p-smith) [inline data-reconciliation-app/Makefile:17]:
```suggestion
test-docker-virtual: build-virtual ## 🧪 Test the Data Reconciliation Application in a Docker sandbox
```

File: data-reconciliation-app/test/test.sh [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+cd ${certificate_dir}
+
+echo "Starting Test..."
+curl -X POST $server/app/log?id=1 --cacert service_cert.pem -H "Content-Type: application/json" --data '{"msg": "Hello Data-reconciliation-app!"}' 
+curl $server/app/log?id=1 --cacert service_cert.pem
+
+echo ""
+echo "Test Completed..."
+echo "OK"
+# kill -9 $sandbox_pid
```
Comment (ross-p-smith) [inline data-reconciliation-app/test/test.sh:78]: you can delete this kill line as we kill the process in a trap

**Review by ross-p-smith — APPROVED** [review]
Thanks for doing this.!

File: data-reconciliation-app/Makefile (hunk 1 above)
Comment (julioalex-rezende) [reply data-reconciliation-app/Makefile:23]: thanks for catching this up

**Pull request merged**